stdlib/public/Glibc/CMakeLists.txt

```diff
@@ -1,46 +1,59 @@
+# When cross-compiling the stdlib on Unix platforms, we'll need a separate
+# glibc for each target.
+foreach(SDK ${SWIFT_SDKS})
+  foreach(arch ${SWIFT_SDK_${SDK}_ARCHITECTURES})
+  string(TOLOWER "${SDK}" sdk)
+  set(output_dir "${SWIFTLIB_DIR}/${sdk}/${arch}/glibc")
 
-# Set correct paths to glibc headers
-set(GLIBC_INCLUDE_PATH "/usr/include")
-if(CMAKE_LIBRARY_ARCHITECTURE)
-  set(GLIBC_ARCH_INCLUDE_PATH "${GLIBC_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
-else()
-  set(GLIBC_ARCH_INCLUDE_PATH "${GLIBC_INCLUDE_PATH}")
-endif()
-if (NOT EXISTS "${GLIBC_ARCH_INCLUDE_PATH}/sys")
-  message(FATAL_ERROR "Glibc headers were not found.")
-endif()
+  # Determine the location of glibc based on the target.
+  if("${SDK}" STREQUAL "LINUX" OR "${SDK}" STREQUAL "FREEBSD")
+    set(GLIBC_INCLUDE_PATH "/usr/include")
+    if(CMAKE_LIBRARY_ARCHITECTURE)
+      set(GLIBC_ARCH_INCLUDE_PATH "${GLIBC_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
+    else()
+      set(GLIBC_ARCH_INCLUDE_PATH "${GLIBC_INCLUDE_PATH}")
+    endif()
+  endif()
 
-# Generate module.map
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-configure_file(module.map.in "${CMAKE_CURRENT_BINARY_DIR}/module.map" @ONLY)
-endif()
-if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
-configure_file(module.freebsd.map.in "${CMAKE_CURRENT_BINARY_DIR}/module.map" @ONLY)
-endif()
+  # Verify that the location is valid.
+  if (NOT EXISTS "${GLIBC_ARCH_INCLUDE_PATH}/sys")
+    message(FATAL_ERROR "Glibc headers were not found at path: ${GLIBC_INCLUDE_PATH}")
+  endif()
 
-add_custom_command_target(unused_var
+  # Configure the modulemap based on the target. Each platform needs to
+  # reference different headers, based on what's available in their glibc.
+  set(modulemap_path "${CMAKE_CURRENT_BINARY_DIR}/${sdk}/${arch}/module.map")
+  if("${SDK}" STREQUAL "FREEBSD")
+    configure_file(module.freebsd.map.in "${modulemap_path}" @ONLY)
+  else()
+    configure_file(module.map.in "${modulemap_path}" @ONLY)
+  endif()
+
+  add_custom_command_target(unused_var
     COMMAND
         "${CMAKE_COMMAND}" "-E" "make_directory" "${output_dir}"
     COMMAND
         "${CMAKE_COMMAND}" "-E" "copy_if_different"
-        "${CMAKE_CURRENT_BINARY_DIR}/module.map"
+        "${modulemap_path}"
         "${output_dir}/module.map"
-    CUSTOM_TARGET_NAME "copy_glibc_module"
+    CUSTOM_TARGET_NAME "copy_${sdk}_${arch}_glibc_module"
     OUTPUT "${output_dir}/module.map" "${output_dir}"
-    DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/module.map"
+    DEPENDS "${modulemap_path}"
     COMMENT "Copying Glibc module to ${output_dir}")
 
-swift_install_in_component(stdlib
+  swift_install_in_component(stdlib
     FILES "${output_dir}/module.map"
-    DESTINATION "lib/swift/glibc")
+    DESTINATION "${output_dir}")
 
-add_swift_library(swiftGlibc IS_SDK_OVERLAY
+  add_swift_library(swiftGlibc IS_SDK_OVERLAY
     ${sources}
-    FILE_DEPENDS copy_glibc_module "${output_dir}"
+    FILE_DEPENDS "copy_${sdk}_${arch}_glibc_module" "${output_dir}"
```

Comment:
We usually use target suffixes instead.

```CMake
set(VARIANT_SUFFIX "-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-${ARCH}")
```

... and append that to target names.